python/cudf/cudf/core/join/join.py

```diff
@@ -72,128 +139,224 @@ def __init__(
             Left and right suffixes specified together, unpacked into lsuffix
             and rsuffix.
         """
+        self._validate_merge_params(
+            lhs,
+            rhs,
+            on=on,
+            left_on=left_on,
+            right_on=right_on,
+            left_index=left_index,
+            right_index=right_index,
+            how=how,
+            lsuffix=lsuffix,
+            rsuffix=rsuffix,
+            suffixes=suffixes,
+        )
+
         self.lhs = lhs
         self.rhs = rhs
+        self.on = on
+        self.left_on = left_on
+        self.right_on = right_on
         self.left_index = left_index
         self.right_index = right_index
-        self.method = method
-        self.sort = sort
-
-        # check that the merge is valid
-
-        self.validate_merge_cfg(
-            lhs,
-            rhs,
-            on,
-            left_on,
-            right_on,
-            left_index,
-            right_index,
-            how,
-            lsuffix,
-            rsuffix,
-            suffixes,
-        )
         self.how = how
-        self.preprocess_merge_params(
-            on, left_on, right_on, lsuffix, rsuffix, suffixes
-        )
+        self.sort = sort
+        self.lsuffix = lsuffix
+        self.rsuffix = rsuffix
+        self.suffixes = suffixes
```

Comment:
I think `self.suffixes` might be redundant/unused

Reply:
Fixed to handle suffixes exclusively via `suffixes`.